iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -279,6 +338,51 @@ public static Anonymizer write(Network n, Path xmlFile) {
         return write(n, new ExportOptions(), xmlFile);
     }
 
+    public static Anonymizer write(Network network, ExportOptions options, DataSource dataSource, String dataSourceExt) throws IOException {
+        try (OutputStream osb = dataSource.newOutputStream(dataSource.getBaseName() + "" + dataSourceExt, false);
+             BufferedOutputStream bosb = new BufferedOutputStream(osb)) {
+
+            if (options.getMode() == IidmImportExportMode.UNIQUE_FILE) {
```

Comment:
I don't understand  this. Why in case of unique file, anonymization dictionnary is written before exporting the network?

Reply:
anonymization dictionnary is not written before exporting the network, if we're here (IidmImportExportMode.UNIQUE_FILE) we have a return statement after writing the anonymizer.

Reply:
Ok I see